src/connector/src/sink/google_pubsub.rs

```diff
@@ -0,0 +1,331 @@
+                "Google Pub/Sub sink only support `Json` sink encode"
+            )));
+        }
+
+        let conf = &self.config;
+        if matches!((&conf.emulator_host, &conf.credentials), (None, None)) {
+            return Err(SinkError::GooglePubSub(anyhow!(
+                "Configure at least one of `emulator_host` and `credentials` in the Google Pub/Sub sink"
+            )));
+        }
+
+        Ok(())
+    }
+
+    async fn new_log_sinker(&self, _writer_param: SinkWriterParam) -> Result<Self::LogSinker> {
+        Ok(
+            GooglePubSubSinkWriter::new(
+                self.config.clone(),
+                self.schema.clone(),
+                &self.format_desc,
+            )
+            .await?
+            .into_log_sinker(usize::MAX),
+        )
+    }
+}
+
+impl TryFrom<SinkParam> for GooglePubSubSink {
+    type Error = SinkError;
+
+    fn try_from(param: SinkParam) -> std::result::Result<Self, Self::Error> {
+        let schema = param.schema();
+        let config = GooglePubSubConfig::from_hashmap(param.properties)?;
+        Ok(Self {
+            config,
+            schema,
+            name: param.sink_name,
+            format_desc: param
+                .format_desc
+                .ok_or_else(|| SinkError::Config(anyhow!("missing FORMAT ... ENCODE ...")))?,
+            is_append_only: param.sink_type.is_append_only(),
+        })
+    }
+}
+
+// ===
+
+struct GooglePubSubPayloadWriter {
+    topic: Topic,
+    publisher: Publisher,
+}
+
+impl GooglePubSubSinkWriter {
+    pub async fn new(
+        config: GooglePubSubConfig,
+        schema: Schema,
+        format_desc: &SinkFormatDesc,
+    ) -> Result<Self> {
+        config.initialize_env();
+
+        let client_config = ClientConfig {
+            endpoint: config.endpoint,
+            project_id: Some(config.project_id),
+            ..Default::default()
+        }
+        .with_auth()
+        .await
+        .map_err(|e| SinkError::GooglePubSub(anyhow!(e)))?;
+        let client = Client::new(client_config)
+            .await
+            .map_err(|e| SinkError::GooglePubSub(anyhow!(e)))?;
+
+        let topic = async {
+            let topic = client.topic(&config.topic);
+            if !topic.exists(None).await? {
+                topic.create(None, None).await?;
+            }
+            Ok(topic)
+        }
+        .await
+        .map_err(|e: Status| SinkError::GooglePubSub(anyhow!(e)))?;
+
+        let timestamptz_mode = TimestamptzHandlingMode::from_options(&format_desc.options)?;
+        let encoder = match format_desc.format {
+            SinkFormat::AppendOnly => match format_desc.encode {
+                SinkEncode::Json => RowEncoderWrapper::Json(JsonEncoder::new(
+                    schema,
+                    None,
+                    DateHandlingMode::FromCe,
+                    TimestampHandlingMode::Milli,
+                    timestamptz_mode,
+                    TimeHandlingMode::Milli,
```

Comment:
Why not use their string form by default? It is quite confusing to see `"date_of_birth": 730119, "daily_meeting_time": 39600000`.

Or another question, why not reuse `SinkFormatterImpl::new`?